library/common/types/c_types.h

```diff
@@ -140,6 +140,20 @@ typedef struct {
   int32_t attempt_count;
 } envoy_error;
 
+/**
+ * Contains internal HTTP stream metrics, context, and other details.
+ *
+ * Note these values may change over the lifecycle of a stream.
+ */
+typedef struct {
+  // An internal identifier for the stream.
+  uint64_t stream_id;
+  // An internal identifier for the connection carrying the stream.
+  uint64_t connection_id;
+  // The number of internal attempts to carry out a request/operation.
+  uint64_t attempt_count;
```

Comment:
@carloseltuerto anything of yours you think should be exposed on a per-call basis?

Reply:
Yep, the idea is that this will be the place to do it. :)

Reply:
We can add that.

Reply:
Note, I would call this "L4" bytes, since it includes bytes that an observer of  the L4 stream would see (headers, post-L7 encoding of data), but an L7 observer would not.